### PR TITLE
Usability issues apply button

### PR DESCRIPTION
**Pull request by StefanBabukov**
# Background
#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-555
#### Link to staging deployment URL 
https://ui-pa8qc38qq3ry6b4quhd59hpbjc.scp-staging.biomage.net/
#### Anything else the reviewers should know about the changes here

# Changes
When changing settings, update settings actions were dispatched and the plot was refreshing without pressig the apply button.
### Code changes
The changes are saved in an object called 'changes' and when the apply button is pressed, the state is changed to the 'changes' object.
# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [x] Tested locally with Inframock
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

**Comment by cosa65** [comment]
There are merge conflicts, also could you deploy to staging?

**Comment by cosa65** [comment]
For 'clustering method' we should show a message about Leiden and SLM not being enabled for now, but instead it shows the same message about cosine.

**Comment by cosa65** [comment]
This is just and idea, do you think we could, instead of showing the tooltip on the `Select`, show it for the `Option`s that are disabled?

**Comment by ivababukova** [comment]
Screenshot 2021-02-23 at 11 12 37
I looked at staging, everything looks good apart from this one in the screenshot -- the tooltip in this case should say Leiden or SLM and also should appear only when the user hovers over either of them. Right now it appears even when I hover over Louvain.

File: src/components/data-processing/ConfigureEmbedding/CalculationConfig.jsx [hunk 1]
```diff
@@ -81,115 +101,116 @@ const CalculationConfig = (props) => {
...
 
-        />
-      </Form.Item>
-    </>
-  );
+            onBlur={(e) => setMinimumDistance(e.target.value)}
+          />
+        </Form.Item>
+        <Form.Item label='Distance metric:'>
+          <Tooltip title='Cosine metric is going to be supported on a future version of the platform.'>
```
Comment (ivababukova) [inline src/components/data-processing/ConfigureEmbedding/CalculationConfig.jsx]: this is the place that needs fixing according to my previous comment with the screenshot

**Comment by StefanBabukov** [comment]
> This is just and idea, do you think we could, instead of showing the tooltip on the `Select`, show it for the `Option`s that are disabled?

When you wrap a single option with a tooltip in antd it just doesn't display the option.

File: src/components/data-processing/ConfigureEmbedding/CalculationConfig.jsx [hunk 2]
```diff
@@ -150,15 +150,18 @@ const CalculationConfig = (props) => {
...
-              <Option value='cosine' disabled>Cosine</Option>
-            </Select>
-          </Tooltip>
+          <Select
+            value={umap.distanceMetric}
+            onChange={(value) => setDistanceMetric(value)}
+          >
+            <Option value='euclidean'>Euclidean</Option>
+            <Option value='cosine' disabled>
+              {' '}
```
Comment (cosa65) [inline src/components/data-processing/ConfigureEmbedding/CalculationConfig.jsx:159]: why ar ethese added?

File: src/components/data-processing/ConfigureEmbedding/CalculationConfig.jsx [hunk 3]
```diff
@@ -243,18 +246,28 @@ const CalculationConfig = (props) => {
...
+              <Option value='louvain'>Louvain</Option>
+              <Option value='leiden' disabled>
+                {' '}
+                <Tooltip title='Leiden metric is going to be supported on a future version of the platform.'>
+                  Leiden
+                </Tooltip>
+
+              </Option>
+              <Option value='slm' disabled>
+                {' '}
```
Comment (cosa65) [inline src/components/data-processing/ConfigureEmbedding/CalculationConfig.jsx]: This too

**Review by cosa65 — APPROVED** [review]

**Pull request merged**